### ISSUE TITLE
indicator disappears from certain camera angles/distances with shaders

### ISSUE THREAD

**Issue by Avera9eJoe** [shedaniel/LightOverlay-Fabric#5]
I noticed that running with any shaders makes this mod very difficult to use as the lines disappear at different angles or distances away, and they also seem to have shader _lighting_ applied to it, I.E. they're really dim. Is there a fix for this I'm not seeing, or is this not meant to be used with shaders?

**Comment by shedaniel** [comment]
Any pictures?

**Comment by Web4Guru** [comment]
Is this a "definitely won't fix" or would pictures and other info help?

Thanks for creating the mod though, much appreciated.

**Comment by shedaniel** [comment]
OptiFine related bugs are hard to fix. But I still want to see a picture.

**Comment by Avera9eJoe** [comment]
The newest updates seem to have fixed the camera angles bug, but there is another issue with certain shaders making the overly very dark:
[Image of LightOverlay on with no shaders](https://i.imgur.com/YDWdPUI.png)
[Image of LightOverlay on with sildur's enhanced default](https://i.imgur.com/Nzxh4Lj.png)
Its almost as if something in the pack is being drawn on top of the light overlay. I noticed that shaders also make the outlines caused from structure blocks to get very dim as well. I think this issue is bigger than just LightOverlay though, so it isn't anything you have to worry about.

**Comment by shedaniel** [comment]
How about the vanilla included chunk borders, are they dim as well? Light Overlay is similar to that.

**Comment by Web4Guru** [comment]
I tested the chunk borders and it has the exact same problem.  I don't use them that often and didn't notice before, but it does the exact same thing.  Optifine internal shaders look fine, so maybe it's more on the individual shader developers.  I've only tested Sildurs shaders.  If you're interested, I could try some others.

I'm using all the latest versions (see below).  Here are screenshots.  These were all taken from the same position with slight adjustments to the view angle.  While walking around, the lines flicker between the black lines, the red lines, and nothing.  I say "flicker" but it's angle dependent and often stays as one color for a second or two as you're walking around.

![Optifine Internal Shaders](https://user-images.githubusercontent.com/8009116/60750577-fdd08c00-9f6f-11e9-9b58-c924d0e8e4a7.png)
![Sildurs Enhanced Default Shaders With Missing Overlay](https://user-images.githubusercontent.com/8009116/60750578-fdd08c00-9f6f-11e9-912d-e28b04c98157.png)
![Sildurs Enhanced Default With All Black Lines](https://user-images.githubusercontent.com/8009116/60750579-fdd08c00-9f6f-11e9-85b2-05e2e4bbd325.png)
![Sildurs Enhanced Default With All Red Lines](https://user-images.githubusercontent.com/8009116/60750580-fdd08c00-9f6f-11e9-9630-579943909a81.png)
![No Shaders](https://user-images.githubusercontent.com/8009116/60750581-fdd08c00-9f6f-11e9-8f83-1e4bd6874f8e.png)
![Sildurs Default Shaders - Chunk Border and Light Overlay With Black Lines](https://user-images.githubusercontent.com/8009116/60750862-25c1ee80-9f74-11e9-94e0-1fe9da57eed2.png)

Minecraft 1.14.3
Fabric Loader 0.4.8 build 155-1.14.3
Fabric API 0.3.0 build 188
LightOverlay 1.2.1
ModMenu 1.5.4-85
OptiFabric 0.3.2
OptiFine 1.14.3_HD_U_F1
Windows 10

**Comment by shedaniel** [comment]
Unfortunately I don’t have a good way to fix this as the vanilla chunk borders are affected as well.

**Comment by Web4Guru** [comment]
Yea, I don't think you should worry about it either.  I'd say your mod works perfectly and it's a bug with the shader packs and how MC implements overlays like chunk borders.  I'll see what the shader community has to say.

Thanks again!

**Comment by Avera9eJoe** [comment]
I agree with Web4Guru, I don't think its a big enough issue to worry about. Its good that we're aware of it though.

**Comment by shedaniel** [comment]
That's true, in any case, sorry for any inconvenience caused!

**Comment by JezaGaia** [comment]
Just wanted to let you know that there is easy if not "nice" workaround, if you use the numbers overlay instead of the crosses it works perfectly with all the shaders I tried it with (all versions of SEUS, Sildur and Trilton compatible with 1.15) in 1.15.2 with fabric and optifine
It's not as nice looking but it's perfctly effective, hope this can help others with this issue.

**Comment by zauberparacelsus** [comment]
This issue could perhaps be fixed by changing the crosses to be rendered with whatever method is being used to render the numbers?  I'm assuming it's some kind of decal texture.

**Comment by DranDivine** [comment]
I have some news I wanted to share. I ran Minecraft 1.16.1 with Fabric, Optifine, Optifabric, Fabric API and Mod Menu and Light Overlay, and I found out something very interesting and useful. If you have a beacon with a beam, I actually got the lines to work with shaders as long as the beam was in render distance. I tested it and it worked normally for Sildurs Vibrant Shaders and BSL, but SEUS only showed the overlay on top of tall grass for some reason. So if you run Sildurs or BSL and have a beacon, you can use this with the lines mode when close enough to the beacon to see it.
![2020-07-18_20 45 16](https://user-images.githubusercontent.com/64855209/87859747-c264ed80-c937-11ea-9989-c1ed4f58c27c.png)
![2020-07-18_20 45 49](https://user-images.githubusercontent.com/64855209/87859743-c0029380-c937-11ea-9167-48786a9817e6.png)
![2020-07-18_20 45 47](https://user-images.githubusercontent.com/64855209/87859750-c55fde00-c937-11ea-8cf7-5b8999d13c63.png)

**Comment by douira** [comment]
Not sure if this is related, but Optifine and Sildurs recently fixed a bug where block outlines and break animations were only visible from certain angles.

**Comment by DranDivine** [comment]
That might be it. It has been seen for a while, but I didn't think they
were looking that much into it.

**Comment by douira** [comment]
Tested it, the lines are still buggy with shaders on. They're only visible in a 90° viewing direction range.

**Comment by DranDivine** [comment]
Oh dang it. Oh well. Better keep hoping then for a fix. :p

**Comment by Avera9eJoe** [comment]
The current fix for now is to use the number view instead of lines; for some reason numbers work perfectly fine still

**Comment by DranDivine** [comment]
Well, perfectly is not quite right either. It works most of the time, but
sometimes it messes up and I see like a item template rather than a number,
as well as I can also see them through walls, which is more annoying. But
yeah, they still work.
Either way, I went back to playing without shaders so I can see the
crosses. Lol.

**Comment by Microlife-JasonLi** [comment]
![圖片](https://user-images.githubusercontent.com/45282804/103852622-34e4a800-50e7-11eb-85a9-b614f92fada9.png)
OptiFine_1.16.4_HD_U_G5
light-overlay-5.6.1
Number mode

**Comment by l-2-j** [comment]
Also seems to be an issue with [Iris](https://github.com/IrisShaders/Iris), where both the numbers and crosses aren't visible at all. Wanted to note it here since as opposed to Optifine the source code of Iris seems to be available should someone want to investigate the issue further.

**Comment by coderbot16** [comment]
This is fixed on the side of Iris in https://github.com/IrisShaders/Iris/tree/1.16.x/refactor/bulletproof-state-tracking and this fix will be included in a future Iris release.

![image](https://user-images.githubusercontent.com/22420959/162606655-2e27c214-623f-402e-b404-cf182e3dda0c.png)

Of course, OptiFine compatibility will still be broken, but clearly that is an OptiFine issue, not an issue with Light Overlay.
